openhands/controller/agent_controller.py

```diff
@@ -166,10 +166,11 @@ async def _start_step_loop(self):
 
             await asyncio.sleep(0.1)
 
-    async def on_event(self, event: Event):
+    async def on_event(self, event_stream: EventStream, event: Event) -> None:
```

Comment:
Are we sure this is better than before? It's not clear to me, because adding this event_stream _seems_ to allow or at least contemplate that various components - various client code - can access a different event stream than the one they have a reference to. Afaik every subscriber has a reference to a single event stream instance, and it's not obvious to me why that's not ok?

Although, one instance of the event stream should only propagate events to [its own subscribers](https://github.com/All-Hands-AI/OpenHands/blob/2a10ff13741dc47452240787b06def2463d38903/openhands/events/stream.py#L143), so maybe they won't get another...Sorry, it's just not clear to me. Can you perhaps explain a bit how does this design make it better?

Reply:
See comment above!

Reply:
Ah, sorry, I just re-read this and understand the comment. Let me look through. I think it might be helpful to make the refactoring a bit more systematic so we can discuss overall which of the two paradigms is desirable.